qiskit_ibm_runtime/base_primitive.py

```diff
@@ -136,7 +136,7 @@ def _run(self, pubs: Union[list[EstimatorPub], list[SamplerPub]]) -> RuntimeJob:
 
         if self._backend and options_dict["transpilation"]["skip_transpilation"]:
             for pub in pubs:
-                self._backend.check_faulty(pub.circuit)
+                self._backend.check_faulty(pub[0])
```

Comment:
~What's wrong with `pub.circuit`? If the input is of type Pub~ 

If the input type is no longer a Pub, the type hint should be updated.

Reply:
Should i put it as union PubLike's or just `tuple`? (its not as general as pub like). I didn't put the unpacking here in the base class because I didn't want to have to do an if /else for if its an estimator or sampler pub.